cpp/src/traversal/sssp_kernels.cuh

```diff
@@ -236,13 +238,18 @@ __global__ void populate_frontier_and_preds(
                     vec_frontier_candidate[iv] = dst_id;
                     ++naccepted_vertices;
                   }
-
-                  // Add src_id to predecessor in either case if needed
-                  if (predecessors) {
-                    predecessors[dst_id] = src_id;
+                }
```

Comment:
The BC specific flows (here and elsewhere) should ideally be behind a flag or a template param. The additional atomics and data transfers between the different predecessor arrays will likely affect performance.

Reply:
Indeed, I will try to avoid affecting the performance of the core SSSP  with the BC specific modifications